File: ex/5.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;

    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int q;
    cin >> q;

    while (q--) {
        int type;
        cin >> type;

        if (type == 1) {  // Запрос суммы
            int l, r;
            cin >> l >> r;
            l--; r--;  // Переводим в 0-индексацию

            long long sum = 0;
            for (int i = l; i <= r; i++) {
                sum += arr[i];
            }
            cout << sum << "\n";
        }
        else {  // XOR обновление
            int l, r, x;
            cin >> l >> r >> x;
            l--; r--;  // Переводим в 0-индексацию

            for (int i = l; i <= r; i++) {
                arr[i] ^= x;
            }
        }
    }

    return 0;
}

File: ex/5.py
import sys
sys.setrecursionlimit(10**9)
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))
q = int(input())

for _ in range(q):
    query = list(map(int, input().split()))

    if query[0] == 1:  # Запрос суммы
        l, r = query[1] - 1, query[2] - 1
        print(sum(arr[l:r + 1]))

    else:  # XOR обновление
        l, r, x = query[1] - 1, query[2] - 1, query[3]
        for i in range(l, r + 1):
            arr[i] ^= x